openlane_update_tool/action.yml

```diff
@@ -0,0 +1,40 @@
+name: Update OpenLane Tool
+
+inputs:
+  tool_name:
+    description: Name of the tool to update.
+    required: true
+  pdk_root:
+    description: Path to the PDK folder.
+    default: .
+  build_ol:
+    description: Build docker image after tool update.
+    default: true
+  ol_tag:
+    description: Tag to use when building the docker image.
+    default: current
+
+runs:
+  using: composite
+  steps:
+
+    - name: Update tool hash
+      shell: bash
+      run: python3 .github/scripts/update_tools.py ${{ inputs.tool_name }}
+
+    - name: Build docker image for updated tool
+      shell: bash
+      run: |
+        make -C docker PDK_ROOT=${{ inputs.pdk_path }} \
+                       OPENLANE_IMAGE_NAME=efabless/openlane:${{ inputs.ol_tag }} \
+                       OPENLANE_DOCKER_TAG=${{ inputs.ol_tag }} \
+                       build-${{ inputs.tool_name }}
+
+    - name: Build OpenLane docker image with new tool binary
+      if: inputs.build_ol == 'true'
+      shell: bash
+      run: |
+        make -C docker PDK_ROOT=${{ inputs.pdk_path }} \
```

Comment:
how do you ensure that this is getting picked up by `openlane_run`? just the fact that the user will have the `tag`s match?

Reply:
`PDK_ROOT` here AFAIK is not very critical. The main reason to set is that OL fails otherwise.